src/main/java/gov/usgs/cida/pubs/utility/XmlUtils.java

```diff
@@ -0,0 +1,188 @@
+	// ftp://ftp.ncbi.nih.gov/pub/jats/extensions/bits/2.0/
+	private static final String[] PUBS_DTDS = {"BITS-book-part-wrap2.ent", "BITS-book-part2.ent", "BITS-book2.dtd",
+			"BITS-bookcustom-classes2.ent", "BITS-bookcustom-mixes2.ent", "BITS-bookcustom-models2.ent",
+			"BITS-bookcustom-modules2.ent", "BITS-bookmeta2.ent", "BITS-common2.ent", "BITS-embedded-index2.ent",
+			"BITS-index2.ent", "BITS-question-answer2.ent", "BITS-toc-index-nav2.ent", "BITS-toc2.ent",
+			"BITS-xinclude2.ent", "JATS-XHTMLtablesetup1.ent", "JATS-ali-namespace1.ent", "JATS-articlemeta1.ent",
+			"JATS-backmatter1.ent", "JATS-chars1.ent", "JATS-common-atts1.ent", "JATS-common1.ent",
+			"JATS-default-classes1.ent", "JATS-default-mixes1.ent", "JATS-display1.ent", "JATS-format1.ent",
+			"JATS-funding1.ent", "JATS-journalmeta1.ent", "JATS-link1.ent", "JATS-list1.ent", "JATS-math1.ent",
+			"JATS-mathml3-mathmlsetup1.ent", "JATS-mathml3-modules1.ent", "JATS-modules1.ent", "JATS-nlmcitation1.ent",
+			"JATS-notat1.ent", "JATS-para1.ent", "JATS-phrase1.ent", "JATS-references1.ent", "JATS-related-object1.ent",
+			"JATS-section1.ent", "JATS-xmlspecchars1.ent", "mathml3.dtd", "mathml/mmlalias.ent", "mathml/mmlextra.ent",
+			"mathml3-qname1.mod", "xhtml-inlstyle-1.mod", "xhtml-table-1.mod", "iso8879/isobox.ent",
+			"iso8879/isocyr1.ent", "iso8879/isocyr2.ent", "iso8879/isodia.ent", "iso8879/isolat1.ent",
+			"iso8879/isolat2.ent", "iso8879/isonum.ent", "iso8879/isopub.ent", "iso9573-13/isoamsa.ent",
+			"iso9573-13/isoamsb.ent", "iso9573-13/isoamsc.ent", "iso9573-13/isoamsn.ent", "iso9573-13/isoamso.ent",
+			"iso9573-13/isoamsr.ent", "iso9573-13/isogrk3.ent", "iso9573-13/isomfrk.ent", "iso9573-13/isomopf.ent",
+			"iso9573-13/isomscr.ent", "iso9573-13/isotech.ent", "xmlchars/isogrk1.ent", "xmlchars/isogrk2.ent",
+			"xmlchars/isogrk4.ent" };
+
+	private static final List<String> IMAGE_NODE_NAMES = List.of("graphic");
+
+	public static final String xsltResourceDir = "xslt";
+	public static final String pubsStyleSheet = "pubs-html.xsl";
+
+	// TODO: make this a configurable parameter
+	// This is a url for pulling images from SPN
+	public static final String SPN_IMAGE_URL = "https://pubs.usgs.gov/xml_test/Images/";
+
+	/**
+	 * Return HTML derived from the given xml document. This html is generating by applying a pubs xslt
+	 * style sheet. DTD schema validation is not done during the transform, the dtd is still read during the 
+	 * transform to bring in Xml entities as needed.
+	 * 
+	 * @param xmlDocUrl The url to the xml doc to convert to HTML.
+	 * @param context
+	 * @return
+	 * 
+	 * @throws TransformerException
+	 * @throws IOException
+	 * @throws ParserConfigurationException
+	 * @throws SAXException
+	 */
+	public static String getPublicationHtml(String xmlDocUrl, ServletContext context)
+			throws TransformerException, IOException, ParserConfigurationException, SAXException {
+		File xsltDir = new ClassPathResource(xsltResourceDir).getFile();
```

Comment:
When I tried getting the stylesheet directory as a class path resource it did not find it.
Because I think, the xslt directory is in the class path, but not the contents.